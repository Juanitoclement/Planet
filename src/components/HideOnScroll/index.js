import { Slide, useScrollTrigger } from "@material-ui/core";
import PropTypes from "prop-types";
import React from "react";

function HideOnScroll(props) {
  const { children, window } = props;
  const trigger = useScrollTrigger({ target: window ? window() : undefined });

  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  );
}

HideOnScroll.propTypes = {
  children: PropTypes.element.isRequired,
  window: PropTypes.func,
};

export default HideOnScroll;
