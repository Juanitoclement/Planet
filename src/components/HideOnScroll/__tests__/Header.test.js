import { render } from "@testing-library/react";
import HideOnScroll from "../../HideOnScroll";

describe("HideOnScroll", () => {
  test("Should Render HideOnScroll", () => {
    const { container } = render(
      <HideOnScroll window={jest.fn()}>
        <div>Hide On Scroll</div>
      </HideOnScroll>
    );

    expect(container).toMatchSnapshot();
  });
});
