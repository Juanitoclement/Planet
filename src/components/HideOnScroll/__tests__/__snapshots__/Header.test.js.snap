// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideOnScroll Should Render HideOnScroll 1`] = `
<div>
  <div>
    Hide On Scroll
  </div>
</div>
`;
